src/python/WMCore/MicroService/MSPileup/MSPileupData.py

```diff
@@ -129,22 +147,41 @@ def createPileup(self, pdict, rseList):
             return [err.error()]
         return []
 
-    def updatePileup(self, doc, rseList=None, validate=True):
+    def updatePileup(self, doc, rseList=None, validate=True, userDN=''):
         """
         Update pileup data with provided input
 
         :param doc: a dictionary of pieleup data to be updated
         :param rseList: a list of valid RSE names
         :param validate: boolean defining whether the doc needs
                          to be validated or not
+        :param userDN: a user DN used in HTTP PUT request
         :return: list of MSPileupError or empty list
         """
+        self.logger.info("Updating pileup document %s", doc)
         rseList = rseList or []
         pname = doc.get('customName', '')
         spec = {'customName': pname}
         if pname == '':
             pname = doc.get('pileupName', '')
             spec = {'pileupName': pname}
+
+        # add transition record if doc contains new fraction
+        fraction = doc.get('containerFraction', 0)
+        transition = doc.get('transition', [])
+        if fraction > 0:
```

Comment:
it is meaningful because of line 170 `fraction = doc.get('containerFraction', 0)` which provide default value for `containerFraction`. I can remove the default but it will be prompt to error if `containerFraction` is not in a document. I think it is better to be safe.